cluster/deploy.sh

```diff
@@ -62,18 +62,33 @@ $KUBECTL delete customresourcedefinitions vms.kubevirt.io  || :
 # Remove all external facing services
 externalServiceManifests | cluster/kubectl.sh --core delete -f - || :
 
-# Delete everything else
-for i in `ls manifests/*.yaml`; do
-    $KUBECTL delete -f $i --grace-period 0 2>/dev/null || :
-done
+if [[ -n "$DEPLOY_MANIFEST" ]]; then
+    $KUBECTL delete -f $DEPLOY_MANIFEST --grace-period 0 2>/dev/null || :
+
+    externalServiceManifests | cluster/kubectl.sh --core apply -f -
+
+    $KUBECTL create -f $DEPLOY_MANIFEST
+else
+    # Delete everything else
+    for i in `ls manifests/*.yaml`; do
+        $KUBECTL delete -f $i --grace-period 0 2>/dev/null || :
+    done
+
+    sleep 2
 
-sleep 2
+    echo "Deploying ..."
+    externalServiceManifests | cluster/kubectl.sh --core apply -f -
 
-echo "Deploying ..."
-externalServiceManifests | cluster/kubectl.sh --core apply -f -
+    for i in `ls manifests/*.yaml`; do
+        $KUBECTL create -f $i
+    done
+fi
 
-for i in `ls manifests/*.yaml`; do
-    $KUBECTL create -f $i
+echo "Wait until kubevirt pods are running"
+while [ -n "$($KUBECTL get pods --all-namespaces --no-headers | grep -v Running)" ]; do
```

Comment:
Sometimes terminated pods stick around for awhile, which I think would cause this to block for quite some time.

Reply:
Right - I actually experienced this. I wonder what the best way is to handle this.

We could ignore them I guess. Thoughts?

Reply:
i think ignoring terminating is fine. If there's a problem with the new manifests, it will result in an ERROR or CrashLoop.